### PR TITLE
Add check_call_in_cache method to check cache without calling function (as per note in the code). Add some tests.

### PR DESCRIPTION
**Pull request by cottrell**

### CODE REVIEW

**Comment by cottrell** [comment]
Code coverage is showing changes in parallel_backends and logger for some reason. No changes there.

File: joblib/test/test_memory.py [hunk 1]
```diff
@@ -479,6 +479,21 @@ def test_persistence(tmpdir):
     gp(1)
 
 
+def test_check_call_in_cache(tmpdir):
+    for func, Result in zip((MemorizedFunc(f, tmpdir.strpath),
```
Comment (pierreglaser) [inline joblib/test/test_memory.py]: What is the point of the `Result` object here? I don't see it used.

File: joblib/memory.py [hunk 2]
```diff
@@ -782,6 +782,18 @@ def _persist_input(self, duration, args, kwargs, this_duration_limit=0.5):
         return metadata
 
     # XXX: Need a method to check if results are available.
+    def check_call_in_cache(self, *args, **kwargs):
+        """
```
Comment (pierreglaser) [inline joblib/memory.py]: Can you make the docstring follow the [numpy docstring convention](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)? Thanks!

Comment (cottrell) [reply joblib/memory.py]: Fixed the docstring I think.

File: joblib/test/test_memory.py (hunk 1 above)
Comment (cottrell) [reply joblib/test/test_memory.py]: Ah yeah, was just copy/pasting. Cleaned that up now.

Comment (cottrell) [reply joblib/test/test_memory.py]: FYI squashed those into one commit now

**Comment by cottrell** [comment]
Just pushed the wrong branch in case someone was about to do something here ... fixing in one sec.

**Comment by cottrell** [comment]
> Just pushed the wrong branch in case someone was about to do something here ... fixing in one sec.

Squashed again to one commit. And import removal had crept in and the test had not picked it up. I think the risks are now small as it is only one screen of new lines to view in the single commit.

**Comment by aabadie** [comment]
It seems to me that this PR is addressing the same problem as #730. Did you have a look at it first ?

**Comment by cottrell** [comment]
> It seems to me that this PR is addressing the same problem as #730. Did you have a look at it first ?

nope. 20 line diff. can't even remember this. Whoever is managing this repo just chuck whatever is easiest. Nobody getting paid here for this so no hard feelings :)  as long and the functionality gets in.

**Comment by tomMoral** [comment]
This PR seems to have stalled. I think this would be useful. 
It seems that the only missing parts are an entry in the changelog and a mention in the doc. @cottrell are you available to help getting this merged?

**Comment by cottrell** [comment]
@tomMoral just saw this ... I've added something to the doc and the changelog. I'm assuming the pipeline does the auto gen part of the doc stuff.

Rebased on master.

**Comment by cottrell** [comment]
Not sure about those test failures, likely the master I rebased on. I don't think those failures happened before the rebase.

**Comment by danlkv** [comment]
What happened? Why this is not merged?

**Comment by tomMoral** [comment]
I did not notice the update from @cottrell , sorry!
Indeed, we should merge this one. I will rebase on master and try to get this merge asap.

**Comment by tomMoral** [comment]
@aabadie @ogrisel I think this feature is helpful and the proposed solution is nice.
Not so sure about the naming.

File: joblib/memory.py [hunk 3]
```diff
@@ -606,6 +609,21 @@ def __getstate__(self):
...
 
+    def check_call_in_cache(self, *args, **kwargs):
+        """Check if function call is in the memory cache.
+
+        Does not call the function or do any work besides func inspection
+        and arg hashing.
+
+        Returns
+        -------
+        is_called_in_cache: bool
```
Comment (tomMoral) [inline joblib/memory.py:620]:
```suggestion
        is_call_in_cache: bool
```

File: joblib/memory.py [hunk 4]
```diff
@@ -606,6 +609,21 @@ def __getstate__(self):
...
+            whether or not the function has been cached for the input
+            arguments that have been passed.
```
Comment (tomMoral) [inline joblib/memory.py:622]:
```suggestion
            Whether or not the result of the function has been cached
            for the input arguments that have been passed.
```

**Review by tomMoral — COMMENTED** [review]
Some nitpicks.

File: CHANGES.rst [hunk 5]
```diff
@@ -111,6 +114,7 @@ Release 0.14.1
   `pkg_resources` that introduced a performance regression under Windows:
   https://github.com/joblib/joblib/issues/965
 
+
```
Comment (tomMoral) [inline CHANGES.rst:117]: ```suggestion```

**Pull request merged**